### PR TITLE
ref(endpoints): Pass OrganizationMember through `convert_args()`

### PR DESCRIPTION
**Pull request by mgaeta**
In this PR I combine several redefinitions of `_get_member()` into the base endpoint `OrganizationMemberEndpoint`.

### CODE REVIEW

File: src/sentry/api/bases/organizationmember.py
Comment (AniketDas-Tekky) [inline src/sentry/api/bases/organizationmember.py:78]: Are we doing DB roundtrip for each arg in `convert_args`?

Comment (mgaeta) [reply src/sentry/api/bases/organizationmember.py:78]: yes

Comment (mgaeta) [reply src/sentry/api/bases/organizationmember.py:78]: yes

Comment (AniketDas-Tekky) [reply src/sentry/api/bases/organizationmember.py:78]: Outside of the scope of this PR, but should we consolidate the queries?

**Review by AniketDas-Tekky — APPROVED** [review]

File: src/sentry/api/bases/organizationmember.py
Comment (mgaeta) [reply src/sentry/api/bases/organizationmember.py:78]: Probably not because we want the API to be able differentiate between a missing `Organization` and a missing `OrganizationMember`.

File: src/sentry/api/endpoints/organization_invite_request_details.py
Comment (JoshFerge) [inline src/sentry/api/endpoints/organization_invite_request_details.py:51]: isn't member_id always a string?

**Review by JoshFerge — COMMENTED** [review]
could you please wait for someone from the enterprise team to review these changes before merging going forward? appreciate it.

File: src/sentry/api/bases/organizationmember.py
Comment (JoshFerge) [inline src/sentry/api/bases/organizationmember.py:78]: do we use this pattern of `args` and `kwargs` anywhere else in the codebase for filtering? overall I find it harder to read.

**Pull request merged**
